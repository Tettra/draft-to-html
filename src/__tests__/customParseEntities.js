import { convertFromRaw } from 'draft-js'
import convertToHtml from '../'
import mentions from './mentions.json'

const ENTITY_TYPE = {
    AUDIO: 'AUDIO',
    EMBEDLY: 'EMBEDLY',
    FILE: 'FILE',
    GITHUB_ISSUE: 'GITHUB-ISSUE',
    GOOGLE_DOC: 'GOOGLE-DOC',
    IFRAME: 'IFRAME',
    IMAGE: 'IMAGE',
    LINK: 'LINK',
    MENTION: 'MENTION',
    PAGE_REFERENCE: 'PAGE-REFERENCE',
    VIDEO: 'VIDEO',
    HR: 'HR',
    TABLE: 'TABLE',
};


const parseEntity = (entity) => {
  // eslint-disable-next-line
  const { url, display_name, title } = entity.data

  if (['PAGE-REFERENCE', 'GITHUB-ISSUE', 'GOOGLE-DOC'].includes(entity.type)) {
    const el = document.createElement('A')
    el.setAttribute('title', title)
    el.setAttribute('href', url)
    return el
  } else if (entity.type === 'MENTION') {
    const el = document.createElement('A')
    el.setAttribute('title', display_name)
    el.setAttribute('href', url)
    return el
  }
}

describe.only('convertToHtml with custom parseBlock option', () => {
  test('converts mentions with custom parseBlock function', () => {
    const html = convertToHtml(convertFromRaw(mentions), { parseEntity })
    expect(html).toMatchSnapshot()
  })
})
