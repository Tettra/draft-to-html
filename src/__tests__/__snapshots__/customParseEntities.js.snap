// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convertToHtml with custom parseBlock option converts mentions with custom parseBlock function 1`] = `"<p><a title=\\"Notes on Link Building [WIP]\\" href=\\"https://staging.tettra.co/teams/tettra/pages/notes-on-link-building-wip\\">Google</a> <a title=\\"Notes on Link Building [WIP]\\" href=\\"https://staging.tettra.co/teams/tettra/pages/notes-on-link-building-wip\\">Page</a> <a title=\\"Jules\\" href=\\"https://dev.tettra.co/teams/jules/users/1\\">@Jules</a> <a title=\\"Use Redstone for Diff (User ability to view diff of large revisions)\\" href=\\"https://github.com/Tettra/tettra-wiki/issues/2352\\">Github</a></p>"`;
